lib/kubernetes-deploy/kubernetes_resource/persistent_volume_claim.rb

```diff
@@ -3,12 +3,66 @@ module KubernetesDeploy
   class PersistentVolumeClaim < KubernetesResource
     TIMEOUT = 5.minutes
 
+    def init_storage_class(cache)
+      @storage_class = {}
+
+      if @definition.dig("spec", "storageClassName").nil?
+        # if no storage class is defined we try to find the default one
+        # FIXME: This assumes the DefaultStorageClass admission plugin is turned on,
+        # need a way to determine this
+        is_default_class = "storageclass.beta.kubernetes.io/is-default-class"
+
+        default_sc = cache.get_all("StorageClass").select do |sc|
+          sc.dig("metadata", "annotations", is_default_class) == "true"
+        end
+
+        if default_sc.length != 1
+          warn_msg = "Multiple default StorageClasses found. If the DefaultStorageClass " \
+            "admission plugin is turned on, all PVC creation will fail."
+          @logger.warn(warn_msg) if default_sc.length > 1
+          return
+        else
+          # using default storage class
+          sc_name = default_sc[0]["metadata"]["name"]
```

Comment:
Since we have the instance why not set @storage_class and return early.

Reply:
Again - another good pick-up. Not sure what I was thinking here. Fixed with:
https://github.com/Shopify/kubernetes-deploy/pull/469/commits/05f28f9adfc734855fa8f33c233d1967597171a8